onnxruntime/core/providers/cuda/cuda_graph.cc

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cuda/cuda_graph.h"
+
+#include "core/providers/cuda/cuda_common.h"
+#include <cuda_runtime_api.h>
+#include <driver_types.h>
+
+
+namespace onnxruntime {
+
+CUDAGraph::CUDAGraph(cudaStream_t stream) : capture_stream_(stream) {
+#if (defined(CUDA_VERSION) && CUDA_VERSION < 10000)
+  ORT_THROW("CUDA graphs can only be used in Onnxruntime built with CUDA >= 10.0");
+#endif
+  CUDA_CALL_THROW(cudaStreamCreateWithFlags(&replay_stream_, cudaStreamNonBlocking));
```

Comment:
Needs some comments on this file as to why we use different streams for capture and replay ?

Reply:
As each thread maintains a stream, cuda graph does not need to use different streams for capture and replay now. The code has been udpated.